tests/extra_source_and_tests/user_2d_T_shaped_pipe_VIPO_shell/T_shaped_pipe_VIPO_shell.cpp

```diff
@@ -0,0 +1,554 @@
+        addSurfaceProperties(Vec2d(-1.0, 1.0).normalized(), wall_thickness_);
+        addPositionAndVolumetricMeasure(Vecd(DL1 - 0.5 * resolution_ref_, - 0.5 * resolution_ref_), resolution_ref_);
+        addSurfaceProperties(Vec2d(-1.0, -1.0).normalized(), wall_thickness_);
+
+        auto particle_number_mid_surface_02 = int(DH / resolution_ref_);
+        //std::cout << " particle_number_mid_surface_02 = " << particle_number_mid_surface_02 << std::endl;
+        for (int i = 1; i < particle_number_mid_surface_02; i++)
+        {
+            // upper wall
+            Real y1 = DH + (Real(i) + 0.5) * resolution_ref_;
+            Real x = DL1 - 0.5 * resolution_ref_;
+            addPositionAndVolumetricMeasure(Vecd(x, y1), resolution_ref_);
+            Vec2d normal_direction = Vec2d(-1.0, 0);
+            addSurfaceProperties(normal_direction, wall_thickness_);
+            // lower wall
+            Real y2 =  -(Real(i) + 0.5) * resolution_ref_;
+            addPositionAndVolumetricMeasure(Vecd(x, y2), resolution_ref_);
+            addSurfaceProperties(normal_direction, wall_thickness_);
+        }
+
+        auto particle_number_mid_surface_03 = int(1.5 * DH / resolution_ref_);
+        //std::cout << " particle_number_mid_surface_03 = " << particle_number_mid_surface_03 << std::endl;
+        for (int i = 0; i < particle_number_mid_surface_03; i++)
+        {
+            Real y1 = 0.5 * DH + (Real(i) + 0.5) * resolution_ref_;
+            // upper wall
+            Real x = DL + 0.5 * resolution_ref_;
+            addPositionAndVolumetricMeasure(Vecd(x, y1), resolution_ref_);
+            Vec2d normal_direction = Vec2d(1.0, 0);
+            addSurfaceProperties(normal_direction, wall_thickness_);
+            // lower wall
+            Real y2 = 0.5 * DH - (Real(i) + 0.5) * resolution_ref_;
+            addPositionAndVolumetricMeasure(Vecd(x, y2), resolution_ref_);
+            addSurfaceProperties(normal_direction, wall_thickness_);
+        }
+    }
+};
+
+//----------------------------------------------------------------------
+//	Inflow velocity
+//----------------------------------------------------------------------
+struct InflowVelocity
+{
+    Real u_ref_, t_ref_;
+    AlignedBoxShape &aligned_box_;
+
+    template <class BoundaryConditionType>
+    InflowVelocity(BoundaryConditionType &boundary_condition)
+        : u_ref_(U_f), t_ref_(2.0),
+            aligned_box_(boundary_condition.getAlignedBox()){}
+
+    Vecd operator()(Vecd &position, Vecd &velocity)
+    {
+        Vecd target_velocity = Vecd::Zero();
+        
+        Real run_time = GlobalStaticVariables::physical_time_;
+        Real u_ave = run_time < t_ref_ ? 0.5 * u_ref_ * (1.0 - cos(Pi * run_time / t_ref_)) : u_ref_;
+        target_velocity[0] = u_ave;
+        target_velocity[1] = 0.0;
+
+        return target_velocity;
+    }
+};
+
+class TimeDependentAcceleration : public Gravity
```

Comment:
Can you try without time dependent acceleration? I'm not sure if it can be used in T-shaped pipe

Reply:
TimeDependentAcceleration is unnecessary here, so I deleted the execution of this step.